physicalplan/plan.go

```diff
@@ -70,6 +70,8 @@ func newOperator(expr parser.Expr, storage storage.Queryable, mint, maxt time.Ti
 		default:
 			return nil, errors.Wrapf(ErrNotSupportedExpr, "got: %s", t)
 		}
+	case *parser.StringLiteral:
+		return nil, nil
```

Comment:
Hm, what does this do?

Reply:
Without this, any expression of type `StringLiteral` would error as "unsupported expression". 

Ideally, we could handle this case right here by returning an operator that returns `StepVector` with `StepSample` of type string, but that needs some changes to the `model` types. This can be iterated on later! So not complicating things for now and checking this in `instantQuery.Exec` instead. 🙂

Reply:
I see, but why do we need to handle StringLiteral for instant queries?

Reply:
Only instant queries can return strings as per docs: https://prometheus.io/docs/prometheus/latest/querying/api/#instant-queries and PromQL has StringLiteral support: https://prometheus.io/docs/prometheus/latest/querying/basics/#string-literals.

This is also kind of how they implement it: https://github.com/prometheus/prometheus/blob/release-2.38/promql/engine.go#L1546 so tried a similar pattern to support it here 🙂

Reply:
Got it, thanks for the explanation.